macros/cross_db_utils/datediff.sql

```diff
@@ -20,13 +20,35 @@
         cast({{second_date}} as datetime),
         cast({{first_date}} as datetime),
         {{datepart}}
-    ) 
+    )
 
 {% endmacro %}
 
 
 {% macro postgres__datediff(first_date, second_date, datepart) %}
 
-    {{ exceptions.raise_compiler_error("macro datediff not implemented for this adapter") }}
+    {% if datepart in ('year', 'decade', 'century', 'millennium') %}
+        (date_part('{{datepart}}', ({{second_date}})::date) - date_part('{{datepart}}', ({{first_date}})::date))
+    {% elif datepart == 'quarter' %}
+        ({{ dbt_utils.datediff(first_date, second_date, 'year') }} * 4 + date_part('quarter', ({{second_date}})::date) - date_part('quarter', ({{first_date}})::date))
+    {% elif datepart == 'month' %}
+        ({{ dbt_utils.datediff(first_date, second_date, 'year') }} * 12 + date_part('month', ({{second_date}})::date) - date_part('month', ({{first_date}})::date))
+    {% elif datepart == 'day' %}
+        (({{second_date}})::date - ({{first_date}})::date)
+    {% elif datepart == 'week' %}
+        ({{ dbt_utils.datediff(first_date, second_date, 'day') }} / 7 + case when date_part('dow', ({{second_date}})::timestamp) < date_part('dow', ({{first_date}})::timestamp) then 1 else 0 end)
```

Comment:
I've spent far longer than I'd like to admit deciphering this line (and am still left scratching my head). Fortunately, I've tested it pretty thoroughly and I'm convinced it works 👍

Reply:
OK so I found a bit of weirdness. If you do a datediff between:
* First date: Tuesday 4th Feb 2020
* Second date: Monday 3rd Feb 2020 (yes a prior day)
Then the result is `1`, whereas I _think_ the result should be 0.

This is certainly a strange edge case, so I'm not super concerned here. But wanted to flag it just in case. I'm happy to merge it as is though!

Reply:
Ugh, you're right. Thanks for catching that. Turns out postgres _rounds toward 0_ in integer division instead of always down (but casting from `double` to `integer` rounds down; I expected the opposite behavior).

Reply:
I'm so glad you know what's happening here ha. I truly had no idea!